packages/contracts-bedrock/src/universal/StandardBridge.sol

```diff
@@ -352,6 +426,126 @@ abstract contract StandardBridge is Initializable {
+
+    function _initiateBridgeERC20WithPermit2(
+        Permit2SignatureTransferData calldata _signatureTransferData,
+        address _remoteToken,
+        address _from,
+        address _to,
+        uint32 _minGasLimit,
+        bytes memory _extraData
+    )
+        internal
+    {
+        require(
+            !_isOptimismMintableERC20(_signatureTransferData.permit.permitted.token),
+            "StandardBridge: Permit2 bridge must be submitted with a native token"
+        );
+
+        permit2.permitTransferFrom(
+            _signatureTransferData.permit,
+            _signatureTransferData.transferDetails,
+            _from,
+            _signatureTransferData.signature
+        );
+        deposits[_signatureTransferData.permit.permitted.token][_remoteToken] = deposits[_signatureTransferData
+            .permit
+            .permitted
+            .token][_remoteToken] + _signatureTransferData.permit.permitted.amount;
+
+        // Emit the correct events. By default this will be ERC20BridgeInitiated, but child
+        // contracts may override this function in order to emit legacy events as well.
+        _emitERC20BridgeInitiated(
+            _signatureTransferData.permit.permitted.token,
+            _remoteToken,
+            _from,
+            _to,
+            _signatureTransferData.permit.permitted.amount,
+            _extraData
+        );
+
+        _sendFinalizeBridgeERC20Message(
+            _signatureTransferData.permit.permitted.token,
+            _remoteToken,
+            _from,
+            _to,
+            _signatureTransferData.permit.permitted.amount,
+            _extraData,
+            _minGasLimit
+        );
+    }
+
+    function _initiateBatchBridgeERC20WithPermit2(
+        Permit2BatchSignatureTransferData calldata _signatureTransferData,
+        address[] calldata _remoteTokens,
+        address _from,
+        address _to,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    )
+        internal
+    {
+        uint256 numPermitted = _signatureTransferData.permit.permitted.length;
+
+        unchecked {
+            for (uint256 i = 0; i < numPermitted; ++i) {
+                require(
+                    !_isOptimismMintableERC20(_signatureTransferData.permit.permitted[i].token),
+                    "StandardBridge: Permit2 bridge must be submitted with a native token"
+                );
+            }
+        }
+
+        permit2.permitTransferFrom(
+            _signatureTransferData.permit,
+            _signatureTransferData.transferDetails,
+            _from,
+            _signatureTransferData.signature
+        );
+
+        unchecked {
+            for (uint256 i = 0; i < numPermitted; ++i) {
+                deposits[_signatureTransferData.permit.permitted[i].token][_remoteTokens[i]] = deposits[_signatureTransferData
+                    .permit
+                    .permitted[i].token][_remoteTokens[i]] + _signatureTransferData.permit.permitted[i].amount;
```

Comment:
can we confirm this is ok to do in an unchecked loop? And if so, David suggested doing this in a second unchecked loop instead of one big unchecked loop. Should have the same effect but basically if we add some additional logic in the loop sometime in the future, we have a higher likelihood of doing this in the unchecked loop unintentionally when it has this large scope

Reply:
you're right i actually don't think this line should be unchecked. updated